enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,273 @@
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs and plugins available for Kubernetes/OpenShift and related services so that I can discover, install, and list them. If `odo` was made available by a cluster-admin, I could install it using:
+* `oc krew install odo`
+
+I could then interact with `odo`:
+* `oc odo --help`
+
+##### Example
+```text
+$ oc krew search odo
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+odo                  OpenShift Developer CLI     1.0          Not Installed
+```
+
+#### Story 2
+
+As owner of a CLI or plugin, I want to publish it to users of the cluster. I need to create a Plugin CR for my tool, or provide the required information about my tool
+to a cluster-admin for the creation of a CR:
+* Name
+* Short description
+* Long description
+* Caveats
+* Homepage
+* Version
+* Platform/architecture
+* The image:tag (and registry credentials if required)
+* The path within the image where the binary for the given platform/architecture can be found
+
+### API Extensions
+
+A new CRD will be created:
+* API: `config.openshift.io/v1`
+* Kind: `Plugin`
+* Spec:
+  * `shortDescription`
+  * `longDescription`
+  * `caveats`
+  * `homepage`
+  * `version`
+  * `platforms`:
+    * `platform`
+    * `image`
+    * `imagePullSecret`
+    * `files`
+* Status:
+  * `digests`:
+    * `name`
+    * `digest`
+    * `calculated`
+
+### Risks and Mitigations
+
+Distributing undesirable binaries is always a risk. Some mitigations include requiring cluster-admins to maintain the index, and the verification of downloaded
+binaries using SHA256 hashes. Cluster-admins are responsible for publishing only trusted binaries.
+
+## Design Details
+
+Each plugin will provide an image.
+Each plugin is responsible for creating a CR to hold metadata.  The CR will serve to deliver the metadata and description
+of its deliverable binary. The Controller will use CRs to generate an index for the `oc krew search` command, and `oc krew install <name>` will download the binary from the Controller.
+Users will install OpenShift tools that are known compatible with each cluster version through `oc krew`.
+
+A plugin must provide a Plugin CR. The result of this proposal will be:
+* Plugin Custom Resource Definition compatible with `krew`
+* Must work with image registries that require image pull secrets
+* Use `krew` manage plugins made available via CRs
+* A Controller to manage plugins that will serve binaries from images
+* If possible, Controller should be distributed via OLM
+* Controller should be optional, but enabled by default as it will be replacing the existing mechanism which was intented to be temporary
+  * See: https://github.com/openshift/enhancements/pull/922
```

Comment:
Change the link from PR to `"/enhancements/installer/component-selection.md"` since the other PR is almost merged.